components/HeaderNav/HeaderNav.tsx

```diff
@@ -1,83 +1,102 @@
 import { IoClose } from 'react-icons/io5';
 import { IoMenu } from 'react-icons/io5';
+import ToggleTheme from './ToggleTheme';
 
 export default function HeaderNav() {
-  const [isOpen, setIsOpen] = useState(false);
+  const { colorMode } = useColorMode();
+  const [isLargerThan800] = useMediaQuery('(min-width: 800px)');
 
-  const handleToggleMenu = () => setIsOpen(!isOpen);
   return (
     <Flex
       as="nav"
       align="center"
-      justify="space-between"
-      wrap="wrap"
+      gap="4"
       padding="1.5rem"
-      bg="white"
-      color="black"
-      borderBottom="2px solid"
-      borderBottomColor="gray.200"
+      justifyContent={{ base: 'space-between', md: 'initial' }}
+      bg={colorMode === 'dark' ? 'gray.800' : 'white'}
+      boxShadow="0px 5px 5px rgba(0, 0, 0, 0.05)"
     >
-      <Flex align="center" mr={5}>
+      <Flex minWidth="110px">
         <Link href="/" _hover={{ textDecoration: 'none' }}>
           <Text fontSize="2xl" fontWeight="bold" fontFamily="heading">
             Ash Bergs
           </Text>
         </Link>
       </Flex>
-      <Box display={{ base: 'block', md: 'none' }}>
-        <Menu isOpen={isOpen}>
-          <MenuButton
-            as="div"
-            display={{ base: 'block', md: 'none' }}
-            onClick={handleToggleMenu}
-          >
-            <IconButton
-              icon={isOpen ? <IoClose /> : <IoMenu />}
-              variant="outline"
-              aria-label={`Toggle Menu ${isOpen ? 'close' : 'open'}`}
-            />
-          </MenuButton>
-          <MenuList>
-            <MenuItem as={Link} href="/work">
-              Work
-            </MenuItem>
-            <MenuItem as={Link} href="/art">
-              Art
-            </MenuItem>
-            <MenuItem as={Link} href="/about">
-              About
-            </MenuItem>
-            <MenuItem as={Link} href="/contact">
-              Contact
-            </MenuItem>
-          </MenuList>
-        </Menu>
-      </Box>
-      <Flex
-        display={{ base: 'none', md: 'flex' }}
-        width={{ base: 'full', md: 'auto' }}
-        alignItems="center"
-        flexGrow={1}
-        gap="2"
-      >
+
+      {isLargerThan800 && <DesktopNav />}
+      {!isLargerThan800 && <MobileNav />}
+    </Flex>
+  );
+}
+
+const DesktopNav = () => {
+  return (
+    <Flex
+      display={{ base: 'none', md: 'flex' }}
+      alignItems="center"
+      w="full"
+      justifyContent="space-between"
+    >
+      <Flex gap="4" fontSize="xl" fontFamily="heading">
         <Link href="/work">Work</Link>
         <Link href="/art">Art</Link>
         <Link href="/about">About</Link>
         <Link href="/contact">Contact</Link>
       </Flex>
+      <ToggleTheme />
     </Flex>
   );
-}
+};
+
+const MobileNav = () => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleToggleMenu = () => setIsOpen(!isOpen);
+  return (
+    <Flex gap="4" display={{ base: 'flex', md: 'none' }}>
+      <ToggleTheme />
```

Comment:
I don't really love how the ToggleButton is behind the navigation items on desktop, and in front of it on mobile. 

I suspect this is a great accessibility choice - and it shouldn't move... however that really didn't present well. Maybe a better layout for this area will come to me.